sdk/keyvault/azure-keyvault-administration/samples/access_control_operations_async.py

```diff
@@ -0,0 +1,107 @@
+        # We'll provide a friendly role name, and let a unique role definition name (a GUID) be generated for us.
+        print("\n.. Create a role definition")
+        role_name = "customRole"
+        scope = KeyVaultRoleScope.GLOBAL
+        permissions = [KeyVaultPermission(data_actions=[KeyVaultDataAction.CREATE_HSM_KEY])]
+        role_definition = await client.set_role_definition(scope=scope, role_name=role_name, permissions=permissions)
+        print("Role definition '{}' created successfully.".format(role_definition.role_name))
+
+        # Let's update our role definition to allow reading keys, but not allow creating keys.
+        # To update an existing definition, pass the name keyword argument to set_role_definition. This is the unique
+        # name of the role definition, which is stored in KeyVaultRoleDefinition.name.
+        print("\n.. Update a role definition")
+        new_permissions = [
+            KeyVaultPermission(
+                data_actions=[KeyVaultDataAction.READ_HSM_KEY],
+                not_data_actions=[KeyVaultDataAction.CREATE_HSM_KEY]
+            )
+        ]
+        unique_definition_name = role_definition.name
+        updated_definition = await client.set_role_definition(
+            scope=scope, name=unique_definition_name, role_name=role_name, permissions=new_permissions
+        )
+        print("Role definition '{}' updated successfully.".format(updated_definition.role_name))
+
+        # Now let's create a role assignment to apply our role definition to our service principal.
+        # Since we don't provide the name keyword argument to create_role_definition, a unique role assignment name
+        # (a GUID) is generated for us.
+        print("\n.. Create a role assignment")
+        principal_id = os.environ["AZURE_CLIENT_ID"]
+        definition_id = updated_definition.id
+        role_assignment = await client.create_role_assignment(
+            scope=scope, definition_id=definition_id, principal_id=principal_id
+        )
+        print("Role assignment created successfully.")
+
+        # Let's delete the role assignment.
+        print("\n.. Delete a role assignment")
+        await client.delete_role_assignment(scope=scope, name=role_assignment.name)
+        print("Role assignment deleted successfully.")
+
+        # Finally, let's delete the role definiton as well.
+        print("\n.. Delete a role definition")
+        await client.delete_role_definition(scope=scope, name=definition_id)
+        print("Role definition deleted successfully.")
+
+    except HttpResponseError as e:
+        print("\nThis sample has caught an error. {}".format(e.message))
+
+    finally:
+        await credential.close()
+        await client.close()
+
+
+if __name__ == "__main__":
+    try:
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(run_sample())
+        loop.close()
+
+    except Exception as e:
+        print("Top level Error: {}".format(str(e)))
```

Comment:
This makes a crash harder to debug, by suppressing exception details. I'd just let the samples crash -- errors will be obvious then 😝

Reply:
This did seem a little odd to me. Is this kind of error catching what was meant by "exception message parsing" in [this](https://github.com/Azure/azure-sdk-for-python/issues/8405) issue? I didn't see other instances of parsing exception messages after taking a quick glance through our samples

Reply:
No, that was specifically pointing at these handlers' parsing of Key Vault error messages to output a more to the point message like "you need to enable soft-delete". The handlers themselves are a separate antipattern.